### ISSUE TITLE
Change from database_path to databasePath to invoke factories in Illuminate\Database

### ISSUE THREAD

**Issue by bionicmaster** [laravel/lumen-framework#598]
There's a change in Illuminate\Database  that changes database_path for databasePath with a parameter. Due to Lumen databasePath not accepting the parameter does a double instance of migrations

**Comment by ArmeniaH** [comment]
We have similar issue with the DatabaseSeeder.php being included twice and causing fatal error on redeclaring. This is caused when we use factory helper in seeder.

**Comment by ArmeniaH** [comment]
any updates here ?

**Comment by laurencei** [comment]
Try updating to Lumen `5.4.6`

Looks like Taylor fixed it here: https://github.com/laravel/lumen-framework/commit/d196ab911cf7c5e3b5725d36388ee85b50613105

**Comment by ArmeniaH** [comment]
Yes, actually this caused our code to break, because we were overriding the databasePath() function, and it did not have an argument $paths. Now he added this $path, but our override still was not passing any paths, so Eloquent construct was trying to load ALL database folder files instead of only factories

```
public static function construct(Faker $faker, $pathToFactories = null)
    {
        $pathToFactories = $pathToFactories ?: database_path('factories');

        return (new static($faker))->load($pathToFactories);
    }
```
database_path('factories') this line was just returning path to database folder, not factories folder :)

**Comment by laurencei** [comment]
So just to clarify - is it working ok for you now?

Or do you still have issues?

**Comment by ArmeniaH** [comment]
Everything is ok now, we have updated the our override of Application class, where we were redefining databasePath.

We had

```
class UcraftApplication extends Laravel\Lumen\Application
{
    public function __construct($basePath = null)
    {
        parent::__construct($basePath);
    }
    public function path()
    {
        return $this->basePath . DIRECTORY_SEPARATOR . 'ucraft';
    }
    public function databasePath()
    {
                return $this->path() . '/Database';
    }

}
```

and changed to

```
class UcraftApplication extends Laravel\Lumen\Application
{
    public function __construct($basePath = null)
    {
        parent::__construct($basePath);
    }
    public function path()
    {
        return $this->basePath . DIRECTORY_SEPARATOR . 'ucraft';
    }
    public function databasePath($path = '')
    {
        return $this->path().DIRECTORY_SEPARATOR.'database'.($path ? DIRECTORY_SEPARATOR.$path : $path);
    }

}
```

after which everything started to work fine, thanks to youe comment

**Comment by bionicmaster** [comment]
For me, it was solved with 5.4.6 that Taylor updated after he checked this ticket. thanks
